test/Generalize/leaf.opt

```diff
@@ -0,0 +1,22 @@
+; REQUIRES: solver, synthesis
```

Comment:
since this test is just looking for a bunch of LGTMs out of a two-step process, it's not really possible to see what behavior it's actually testing. please modify it to only run your new tool, and then check that its output is correct without invoking a second tool, so we can look at the test and understand what's going on. also do this to the other test.

Reply:
OK, sounds good.